### DISCUSSION TITLE
Use without global installation?

### DISCUSSION THREAD

**Discussion by devinrhode2** [google/zx#254] [Help needed]
I was interesting doing something like this:
```
yarn install zx --dev
```
Then, inside package.json scripts:
```
"do-some-neat-thing": "zx ./scripts/neat-stuff.js",
```
I think a lot of people may see this as a superior approach

**Comment by devinrhode2** [comment]
I can do a simple test locally to see if this works now - but I'm curious if this project envisions local installs being a well-supported use case long-term.

**Reply by antonmedv** [reply]
This is works now. I didn’t get what is proposed?

**Reply by devinrhode2** [reply]
Firstly, it is just a "political/vision/philosophical" question.

Secondly, I am curious if this is just a flat-out better way to use zx.

If it is quite simply a better way to use zx, then the docs could be changed to reflect the local-only approach.

Something I want to use zx for is cross-platform (windows/linux/mac) git hooks (loaded with huksy). In such an environment, local only is best, as users (other developers commiting in git) will not have to think about the inner-workings at all, and there's no extra setup step we need our developers to do.

Although, I'm really hoping windows devs can just get zsh or bash setup. If that proves to be too difficult, then I think zx could be a sweeter alternative to plain node scripts.

**Comment by antongolub** [comment]
@devinrhode2,

How about `npm_config_yes=true npx zx [...]`?
